explorer/src/lib/components/transaction-details/TransactionDetails.svelte

```diff
@@ -53,8 +54,47 @@
   /** @type {boolean} */
   let isPayloadToggled = false;
 
+  /** @type {boolean} */
+  let isMemoDecoded = false;
+
   $: classes = makeClassName(["transaction-details", className]);
 
+  /**
+   * @param {string} hex
+   * @returns {string}
+   */
+  function checkHexString(hex) {
+    if (!isHex(hex)) {
+      return hex;
+    }
+
+    /**
+     * @returns {Uint8Array}
+     */
+    const hexToBytes = () => {
```

Comment:
Generally, I would extract as a helper on its own, so as that it can be tested properly. 

I wanted to see cases when the hex starts with -0x|0x, I am not sure if these has to be stripped first.